signer/src/testing/transaction_coordinator.rs

```diff
@@ -992,27 +1016,40 @@ where
         let original_test_data = test_data.clone();
 
         let test_data_rc = RefCell::new(test_data);
-        let mut push_block = |parent| {
+        let rng_rc = RefCell::new(rng);
+        let push_block = |parent| {
             let (block, block_ref) = test_data_rc.borrow_mut().new_block(
```

Comment:
Hm, this feels fishy... `rng` implements `Clone`, couldn't you have cloned it to a mutable var in the closure if it was a borrow checker issue?

Reply:
I did originally, but that has it's own issue since you get the same output each time the closure is called, since the RNG is called with the same starting state.

Reply:
I had misunderstood the `Clone`ing behavior of the rng since it's Sync + Send, I thought it would use interior mutability but no it literally branches off of its current state 🤦